### ISSUE TITLE
Multiple firefox windows - FF window on 2nd display is moved to primary display 

### ISSUE THREAD

**Issue by MisterT87** [kangyu-california/PersistentWindows#58]
Hi again Guys 😂

I am normally using 2 firefox windows und 2 displays (one for each).
Moved both FF Windows to the displays and took a snapshot with PW...

But very often (not always) the FF window on 2nd display will be moved to primary monitor - which can be very stressful if u working on it 😁

Additional Info:
- enabled tab preview in taskbar (in firefox settings)
- on both FF windows are multiple tabs (but the error log below is always the same - always webpage title is ofcourse changing)
- both FF windows are normal windows (no private windows)

Firefox version: 82.0.2
Windows 10 Pro 2004
Tested with: PersistentWindows 5.1

Error log: 
[Pastebin.com](https://pastebin.com/xkXBMShJ)

Maybe we can find a solution pls ? Sometimes it drives me crazy 😁😁

greetings

**Comment by kangyu-california** [comment]
Just want to conform, there are 3 kinds of restore in latest release.
1. automatic restore
2. manual restore from disk
3. manual restore snapshot (introduced since release 5.0, taking snapshot by double click  PW  icon, restore snapshot by single click PW icon)

Which type of restore are you referring to?

**Comment by MisterT87** [comment]
first one: automatic restore :) 
Sorry forgot to mention it

**Comment by kangyu-california** [comment]
I did not reproduce the issue. However it seems to be related to issue #57, which is fixed in release 5.2.
Please try and feedback.

Thanks

**Comment by MisterT87** [comment]
@kangyu-california Thanks already updated PW to v. 5.2 - give me 7 days for long time testing :)

**Comment by MisterT87** [comment]
@kangyu-california Sorry, but issue still exists in PW 5.2 :'(

**Comment by kangyu-california** [comment]
Issue reproduced. Will be fixed in 5.4

**Comment by MisterT87** [comment]
@kangyu-california thanks for the hint - but already using this feature ;)
Updated PW to 5.3 and testing...

btw is there a way to support you (monetary) ? :)

**Comment by kangyu-california** [comment]
Thanks for the appreciation, this project should keep open and free.

**Comment by MisterT87** [comment]
@kangyu-california ok, just asking :)

But the issue still exist in PW 5.3 :(
same error but some new infos to reproduce:

I have only this issue with one FF window on 2nd screen - the already opened Firefox window on primary monitor wont be moved on auto restore 🤔

I have this issue also if I move Firefox window to primary monitor - for example: normaly window is minimized on 2nd monitor if I maximize window it will be moved to primary monitor.
But If I move the same window now on the primary monitor manually, minimize it and wait a lil bit - after maximizing in will be moved to 2nd monitor then :/

really strange -.-

As i can say the issue only occurs if Firefox window was minimized and dont touch the window for around 5 minutes - then If I want maximize the firefox window it will be moved after maximizing.
Now when I am trying again to minimize the firefox window on 2nd screen and maximize it again it will be also moved to primary monitor... tried this a few minutes ago and that happened 3 times sequentally - on the 4th time the firefox window wasnt moved anymore

**Comment by kangyu-california** [comment]
It seems this issue is same as #57,

My speculation is the last window position (before minimize operation) is not captured by PersistentWindows due to insufficient waiting time. Usually it takes 3~6 seconds delay to capture (after all windows are settled still)

My suggestion is to try slow down the pace in testing.

**Comment by MisterT87** [comment]
@kangyu-california I'll try it tommorow :)
one question offtopic: need to know: When I am using "capture windows to disk" feature... In which file it will be stored and will this file overwritten if I am replacing old files from newer .zip files (due to PW update) ?

**Comment by kangyu-california** [comment]
.db file location is c:/users/yourID/appdata/local/persistentwindows/, and It is not affected by upgrading PW. You may download litedb studio from git hub to check db file content.

**Comment by MisterT87** [comment]
ah thx - just asked for backup purposes :)

on topic: you was right -> On PW 5.4 - issue only exists when I minize the firefox window too fast. 
After waiting 6secs before minimizing -> window wont be moved after maximizing !

tldr: problem solved ;)

But would be nice if you can write down a hint about this 6s waiting time, on github and maybe also on the "new HTML file - where I can see all CLI options in the future" 
pretty sure it will avoid alot issue postings ;D

**topic can be closed after answering**

**Comment by kangyu-california** [comment]
Release 5.5 improved capture latency down to 1 second, this applies to moving window by mouse, minimize, maximize.

**Comment by MisterT87** [comment]
yep I have seen it - thats why I am reopen this issue again :D 

but be careful with this timing and how often it will (auto) window positions - I dont know how PW stores the positions on auto...
but I am thinking about abit older/slower machines and hard disk drives (mechanical)

just be carefull and again big thanks for support :)

**Comment by MisterT87** [comment]
@kangyu-california the new release of PW 5.5 is working like a charm - no errors anymore from my side !
also have no issues with the "new timer" for storing (auto) window positions...
